packages/scene-composer/src/augmentations/components/three-fiber/viewpoint/ViewCursorWidget.tsx

```diff
@@ -1,76 +1,94 @@
+    const box = new Box3().setFromObject(object);
+    box.getCenter(object.position);
+    object.position.multiplyScalar(-1);
+  }, [ref]);
 
-  useEffect(() => {
-    if (viewCursorRef.current) {
-      viewCursorRef.current.lookAt(cursorLookAt);
-      viewCursorRef.current.rotation.z = camera.rotation.z; // Prevent spinning and always be straight up and down
-    }
-  }, [cursorLookAt]);
+  /* istanbul ignore next */
+  const shape = useMemo(() => {
+    const paths = data.paths;
+    const svgGroup = new Group();
 
-  useEffect(() => {
-    if (viewCursorRef.current) {
-      viewCursorRef.current.traverse((child) => {
-        const mesh = child as THREE.Mesh;
+    paths?.forEach((path) => {
+      const fillColor = path?.userData?.style.fill;
+      if (fillColor !== undefined && fillColor !== 'none') {
+        const material = new MeshBasicMaterial({
+          color: new Color().setStyle(fillColor).convertSRGBToLinear(),
+          opacity: path?.userData?.style.fillOpacity,
+          transparent: true,
+          depthFunc: AlwaysDepth,
+          side: DoubleSide
+        });
+        const shapes = SVGLoader.createShapes(path);
+        shapes.forEach((line) => {
+          const geometry = new ShapeGeometry(line);
+          const mesh = new Mesh(geometry, material);
+          resetObjectCenter(mesh);
+          svgGroup.add(mesh);
+        });
+      }
 
-        if (mesh.material) {
-          if (Array.isArray(mesh.material)) {
-            mesh.material.forEach((material) => {
-              material.depthFunc = THREE.AlwaysDepth;
-            });
-          } else {
-            (mesh.material as THREE.Material).depthFunc = THREE.AlwaysDepth;
+      const strokeColor = path?.userData?.style.stroke;
+      if (strokeColor !== undefined && strokeColor !== 'none') {
+        const material = new MeshBasicMaterial({
+          color: new Color().setStyle(strokeColor).convertSRGBToLinear(),
+          opacity: path?.userData?.style.strokeOpacity,
+          transparent: true,
+          depthFunc: AlwaysDepth,
+          side: DoubleSide
+        });
+        path.subPaths.forEach((childPath) => {
+          const geometry = SVGLoader.pointsToStroke(childPath.getPoints(), path?.userData?.style);
+          if (geometry) {
+            const mesh = new Mesh(geometry, material);
+            resetObjectCenter(mesh);
+            svgGroup.add(mesh);
           }
-        }
-      });
-    }
-  }, [viewCursorRef]);
+        });
+      }
+    });
+    svgGroup.scale.multiplyScalar(0.005);
+    return svgGroup;
+  }, [data]);
 
-  return (
-    <viewCursor
-      ref={viewCursorRef}
-      isSelected={cursorStyle === 'edit'}
-      position={cursorPosition}
-      visible={cursorVisible}
-    >
-      {moveVisual}
-      {editVisual}
-    </viewCursor>
-  );
-};
+  /* istanbul ignore next */
+  useFrame(({ raycaster }) => {
+    const sceneMeshes: Object3D[] = [];
+    scene.traverse((child) => {
```

Comment:
if this is the only place scene is being used you can get this from the state passed into useFrame such as { raycaster, scene }

Reply:
We have to get the raycaster on frame? This seems unfortunate. 

There are 2 instances where we need to do the collision detection here:
1. When the user clicks the mouse, so we know what to attach things to
2. When the mouse moves over an object (so we can adjust the rotation of the cursor)

For the second case, I can see `useFrame` might actually be more efficient, since the mouse can definitely move faster than 60 times/second, but this creates possible lag as the orientation of the cursor can take a few milliseconds to catch up to the actual mouse position. The problem is while we target 60 frames per second, slower machines with weaker graphics cards may not be able to render at 60 fps, and the aforementioned lag is even more pronounced in those situations.

I'd actually prefer for the second scenario, if we added a debounce on the mouse move to make the lag at least consistent.

Reply:
`useFrame` is actually ideal for what we're doing here as this [hook](https://gracious-keller-98ef35.netlify.app/docs/api/hooks/useFrame/) is designed specifically for effects. 

Raycasting based on mouse movement requires having to constantly look at where the mouse is intersecting in order to know how to cast the `svg`. Without it, we can't place the `svg` in the canvas.

If mouse movement wasn't involved then we wouldn't have a need for the `useFrame`.